operators/src/engine/query.rs

```diff
@@ -1,13 +1,45 @@
-use geoengine_datatypes::primitives::{BoundingBox2D, SpatialResolution, TimeInterval};
+use geoengine_datatypes::primitives::{
+    BoundingBox2D, SpatialPartition, SpatialPartitioned, SpatialResolution, TimeInterval,
+};
 
-/// A spatio-temporal rectangle for querying data
+/// A spatio-temporal rectangle for querying data with a bounding box
 #[derive(Copy, Clone, Debug, PartialEq)]
-pub struct QueryRectangle {
+pub struct VectorQueryRectangle {
     pub bbox: BoundingBox2D,
     pub time_interval: TimeInterval,
     pub spatial_resolution: SpatialResolution,
 }
 
+/// A spatio-temporal rectangle for querying data with a spatial partition
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct RasterQueryRectangle {
+    pub partition: SpatialPartition,
+    pub time_interval: TimeInterval,
+    pub spatial_resolution: SpatialResolution,
+}
```

Comment:
1. I would'nt call it raster/vector, since we don't want to create a PlotQueryRectangle, I guess. So rather Partitioned and Bbox or something like this.
2. We could also say QueryRectangle<B: BoxShaped>, then we can have the rest in common.
3. We could then have only aliases for our types.

Reply:
1. I find it very convenient that it's clear by the name which QueryRectangle to use. We can have a PlotQueryRectangle that is a type alias for VectorQueryRectangle
2. would be possible but we wouldn't save a lot. I think it's better if they are separate types especially because it's possible to turn a vectorQueryRectangle into a RasterQueryRectangle but not vice versa.

Reply:
I guess we have to discuss this.